rskj-core/src/main/java/co/rsk/config/NodeCliOptions.java

```diff
@@ -41,6 +52,62 @@ public Config withConfig(Config config, String configValue) {
             return config.withValue(SystemProperties.PROPERTY_BASE_PATH, ConfigValueFactory.fromAnyRef(configValue));
         }
     },
+    SYNC_MODE("sync-mode", true) {
+        @Override
+        public Config withConfig(Config config, String configValue) {
+            if(configValue.equalsIgnoreCase("full") || configValue.equalsIgnoreCase("snap")){
+                return config.withValue(RskSystemProperties.PROPERTY_SYNC_MODE, ConfigValueFactory.fromAnyRef(configValue));
+            }else {
+                throw new IllegalArgumentException("Invalid sync mode: " + configValue + ". The valid options are <full> or <snap>.");
+            }
+        }
+    },
+    SNAP_NODES("snap-nodes", true) {
+        @Override
+        public Config withConfig(Config config, String configValue) {
+            try {
+                List<ConfigObject> snapConfigObjects = Arrays.stream(configValue.substring(1, configValue.length() - 1).split(","))
+                        .map(String::trim)
+                        .filter(s -> !s.isEmpty())
+                        .map(this::createConfigObjectFromNode)
+                        .collect(Collectors.toList());
+
+                if(!snapConfigObjects.isEmpty()) {
+                    ConfigValue snapConfigValue = ConfigValueFactory.fromIterable(snapConfigObjects);
+                    return config.withValue(RskSystemProperties.PROPERTY_SNAP_NODES, snapConfigValue);
+                }
+                return config;
+            } catch (Exception e) {
+                e.printStackTrace();
+                return config;
+            }
+        }
+        private ConfigObject createConfigObjectFromNode(String snapNode) {
+            try {
+                URI snapURI = new URI(snapNode);
```

Comment:
in cli we expect value to be in this format: `enode://PUBKEY@HOST:PORT`. So we could simply make a ConfigObject in the form of: `{ "url": "<value>" }` without a need to parse it, wdyt?

Reply:
Got it, that makes sense. I’ll simplify the method to directly wrap the URI string in a ConfigObject without parsing it into components

Just to make sure, should we still include some validation to check if the URL is correctly formatted before accepting it? It might help catch any errors early and make sure everything is in the expected format. Let me know what you think

Reply:
solved, removed the substring. The flag should be formatted as `--snap-nodes=<url1>`` --snap-nodes=<url2>`, with each URL specified separately

Reply:
solved, implemented you suggestion [here](https://github.com/rsksmart/rskj/pull/2653/files#diff-2936d53bde166899a1f013108ef53a23fa931a04c7f06dae1df7f4c309a7ce12R92)